internal/controllers/operator_controller_test.go

```diff
@@ -1048,6 +1056,223 @@ func verifyConditionsInvariants(op *operatorsv1alpha1.Operator) {
 	}
 }
 
+func TestOperatorUpgrade(t *testing.T) {
+	// bootstrapping test environment
+	var err error
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			filepath.Join("..", "..", "config", "crd", "bases"),
+			filepath.Join("..", "..", "testdata", "crds")},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	cfg, err = testEnv.Start()
+	require.NoError(t, err)
+	assert.NotNil(t, cfg)
+
+	sch = runtime.NewScheme()
+	require.NoError(t, operatorsv1alpha1.AddToScheme(sch))
+	require.NoError(t, rukpakv1alpha1.AddToScheme(sch))
+
+	cl, err = client.New(cfg, client.Options{Scheme: sch})
+	require.NoError(t, err)
+	assert.NotNil(t, cl)
```

Comment:
Given that we want to get rid of ginkgo tests (see #189) I decided to add a new test for updates instead of expanding existing ginkgo based test (above in this file).

---

Now the issue is - I had to include all this boilerplate into the test and do clean up in each subtest (see `DeleteAllOf` in `defer`s below). For ginkgo we have [`BeforeSuite` here](https://github.com/operator-framework/operator-controller/blob/7d53835f38b8092476fb92b87da4be14567cb527/internal/controllers/suite_test.go#L56-L78) and `AfterSuite`.

Wondering if we can do it differently. Any suggestions?

I considered a few things:
* Adding this stuff into `init`, but I think cleanup will be tricky. We will have to clean up in each test/sub test?
* I was hoping to use controller-runtime's fake client instead, but it doesn't support `Patch` which is being used in `OperatorReconciler`.

Reply:
#473 was merged to address boilerplate related to `envtest`